commit 47b0b1346cf1c00968ab0b3181e943474987f91f
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Tue Aug 27 15:39:25 2024 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Aug 27 15:39:25 2024 -0400

    Fixed the SubprocessLauncher missing app_custom_folder in the PythonPath. (#2857)
---
 nvflare/private/fed/client/client_executor.py | 4 ++--
 nvflare/private/fed/server/server_engine.py   | 9 +++++++--
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/nvflare/private/fed/client/client_executor.py b/nvflare/private/fed/client/client_executor.py
index c775f7b8..8830cdbf 100644
--- a/nvflare/private/fed/client/client_executor.py
+++ b/nvflare/private/fed/client/client_executor.py
@@ -28,7 +28,7 @@ from nvflare.fuel.f3.cellnet.core_cell import FQCN
 from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey, ReturnCode
 from nvflare.fuel.utils.config_service import ConfigService
 from nvflare.private.defs import CellChannel, CellChannelTopic, JobFailureMsgKey, new_cell_message
-from nvflare.private.fed.utils.fed_utils import get_return_code
+from nvflare.private.fed.utils.fed_utils import add_custom_dir_to_path, get_return_code
 from nvflare.security.logging import secure_format_exception, secure_log_traceback
 
 from .client_status import ClientStatus, get_status_message
@@ -172,7 +172,7 @@ class ProcessExecutor(ClientExecutor):
         """
         new_env = os.environ.copy()
         if app_custom_folder != "":
-            new_env["PYTHONPATH"] = new_env.get("PYTHONPATH", "") + os.pathsep + app_custom_folder
+            add_custom_dir_to_path(app_custom_folder, new_env)
 
         command_options = ""
         for t in args.set:
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index f9f3d4cb..65393b7e 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -56,7 +56,12 @@ from nvflare.private.admin_defs import Message, MsgHeader
 from nvflare.private.defs import CellChannel, CellMessageHeaderKeys, RequestHeader, TrainingTopic, new_cell_message
 from nvflare.private.fed.server.server_json_config import ServerJsonConfigurator
 from nvflare.private.fed.server.server_state import ServerState
-from nvflare.private.fed.utils.fed_utils import get_return_code, security_close, set_message_security_data
+from nvflare.private.fed.utils.fed_utils import (
+    add_custom_dir_to_path,
+    get_return_code,
+    security_close,
+    set_message_security_data,
+)
 from nvflare.private.scheduler_constants import ShareableHeader
 from nvflare.security.logging import secure_format_exception
 from nvflare.widgets.info_collector import InfoCollector
@@ -237,7 +242,7 @@ class ServerEngine(ServerEngineInternalSpec):
     ):
         new_env = os.environ.copy()
         if app_custom_folder != "":
-            new_env["PYTHONPATH"] = new_env.get("PYTHONPATH", "") + os.pathsep + app_custom_folder
+            add_custom_dir_to_path(app_custom_folder, new_env)
 
         listen_port = open_ports[1]
         if snapshot:
